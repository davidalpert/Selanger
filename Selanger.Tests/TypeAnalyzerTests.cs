using System;
using System.IO;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;
using Selanger.Core;

namespace Selanger.Tests
{
    [TestFixture]
    public class TypeAnalyzerTests
    {
        [Test]
        public void CanGenerateTypeReport()
        {
            var asm = @".\nunit.framework.dll";
            var file = new FileInfo(asm);
            var reporter = new TypeAnalyzer();
            var result = reporter.Analyze(file);
            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3},{4}", 
                                                            x.Namespace,
                                                            x.Name, 
                                                            x.FileDirectory, 
                                                            x.FileName,
                                                            x.FileVersion
                                                           ));
        }

        [Test]
        public void CanGenerateTypeReportForListOfAssemblies()
        {
            var files = Directory.GetFiles(".", "*.dll",SearchOption.AllDirectories)
                                .Take(3)
                                .Select(f => new FileInfo(f))
                                .ToArray();

            var reporter = new TypeAnalyzer();

            var result = reporter.Analyze(files);

            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3},{4}", 
                                                            x.Namespace,
                                                            x.Name, 
                                                            x.FileDirectory, 
                                                            x.FileName,
                                                            x.FileVersion
                                                           ));
        }

        [Test]
        public void CanGenerateNamespaceReport()
        {
            var asm = @".\nunit.framework.dll";
            var file = new FileInfo(asm);
            var reporter = new TypeAnalyzer();
            var result = reporter.SummarizeNamespaces(file);
            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3}", 
                                                            x.AssemblyName, 
                                                            x.AssemblyVersion,
                                                            x.Namespace, 
                                                            x.Count
                                                           ));
        }

        [Test]
        public void CanGenerateNamespaceReportForListOfAssemblies()
        {
            var files = Directory.GetFiles(".", "*.dll",SearchOption.AllDirectories)
                                .Take(3)
                                .Select(f => new FileInfo(f))
                                .ToArray();

            var reporter = new TypeAnalyzer();

            var result = reporter.SummarizeNamespaces(files);

            Approvals.VerifyAll(result, x => String.Format("{0},{1},{2},{3}", 
                                                            x.Namespace, 
                                                            x.Count,
                                                            x.AssemblyName, 
                                                            x.AssemblyVersion
                                                           ));
        }
    }
}
