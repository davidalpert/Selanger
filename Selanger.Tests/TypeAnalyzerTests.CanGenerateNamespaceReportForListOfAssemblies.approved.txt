,1,ApprovalTests.dll,3.0.0.0
ApprovalTests,11,ApprovalTests.dll,3.0.0.0
ApprovalTests.Approvers,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.Asp,6,ApprovalTests.dll,3.0.0.0
ApprovalTests.Asp.Mvc,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.Combinations,34,ApprovalTests.dll,3.0.0.0
ApprovalTests.Core,9,ApprovalTests.dll,3.0.0.0
ApprovalTests.Core.Exceptions,3,ApprovalTests.dll,3.0.0.0
ApprovalTests.Email,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.Events,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.Html,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.Namers,5,ApprovalTests.dll,3.0.0.0
ApprovalTests.Persistence,2,ApprovalTests.dll,3.0.0.0
ApprovalTests.Persistence.DataSets,2,ApprovalTests.dll,3.0.0.0
ApprovalTests.Persistence.EntityFramework,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.Persistence.EntityFramework.Version5,2,ApprovalTests.dll,3.0.0.0
ApprovalTests.Persistence.NHibernate,1,ApprovalTests.dll,3.0.0.0
ApprovalTests.RdlcReports,5,ApprovalTests.dll,3.0.0.0
ApprovalTests.Reporters,45,ApprovalTests.dll,3.0.0.0
ApprovalTests.StackTraceParsers,12,ApprovalTests.dll,3.0.0.0
ApprovalTests.WinForms,5,ApprovalTests.dll,3.0.0.0
ApprovalTests.Wpf,6,ApprovalTests.dll,3.0.0.0
ApprovalTests.Writers,4,ApprovalTests.dll,3.0.0.0
ApprovalTests.Xml,1,ApprovalTests.dll,3.0.0.0
,3,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Asp.Mvc,1,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.CallStack,6,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Persistence,9,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Persistence.Database,5,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Persistence.EntityFramework,7,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Persistence.NHibernate,2,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Reflection,12,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.SimpleLogger,3,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.SimpleLogger.Writers,6,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Utilities,15,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.WinForms,1,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Wpf,3,ApprovalUtilities.dll,3.0.0.0
ApprovalUtilities.Xml,1,ApprovalUtilities.dll,3.0.0.0
NUnit.Framework,86,nunit.framework.dll,2.6.3.13283
NUnit.Framework.Constraints,122,nunit.framework.dll,2.6.3.13283
